OpenRA.Mods.Common/FileFormats/WavReader.cs

```diff
@@ -129,21 +131,25 @@ public WavStreamImaAdpcm(Stream stream, int dataSize, short blockAlign, short ch
 				numBlocks = dataSize / blockAlign;
 				blockDataSize = blockAlign - channels * 4;
 				outputSize = uncompressedSize * channels * 2;
-				predictor = new int[channels];
-				index = new int[channels];
 
-				interleaveBuffer = new byte[channels * 16];
+				blockData = new byte[blockDataSize];
 			}
 
 			protected override bool BufferData(Stream baseStream, Queue<byte> data)
 			{
 				// Decode each block of IMA ADPCM data
 				// Each block starts with a initial state per-channel
+				Span<int> predictor = stackalloc int[channels];
+				Span<int> index = stackalloc int[channels];
+
+				Span<byte> channelData = stackalloc byte[channels * 4];
+				baseStream.ReadBytes(channelData);
+				var cd = 0;
 				for (var c = 0; c < channels; c++)
 				{
-					predictor[c] = baseStream.ReadInt16();
-					index[c] = baseStream.ReadUInt8();
-					baseStream.ReadUInt8(); // Unknown/Reserved
+					predictor[c] = channelData[cd++] << 8 | channelData[cd++];
```

Comment:
```suggestion
					predictor[c] = (short)(channelData[cd++] | channelData[cd++] << 8);
```

This triggered my unit tests for decoding incorrectly, this is the fix

Reply:
Isn't the order of these two `cd++` within the same expression unspecified?

Reply:
> This triggered my unit tests for decoding incorrectly

Nice catch, thank you.

> Isn't the order of these two `cd++` within the same expression unspecified?

You may be thinking of C or C++ where that is the case. C# uses a well-defined order of evaluation, left-to-right for most things: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-associativity